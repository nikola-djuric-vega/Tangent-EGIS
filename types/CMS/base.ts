import {
  AboutPage,
  AllNewsPage,
  ProjectListingPage,
  ProjectPage,
  SiteSettings,
  PageMetadata,
  HomePage,
  InsightsListingPage,
  InsightPage,
  Media,
  NewsDetailPage,
  SectorPage,
  ServicesPage,
  SubSectorPage,
  CountryPage,
  ContinentPage,
  EventDetailPage,
  PDFDownloadPage,
  StandardPage,
  WebinarPage,
  CareerPage,
  CSRPage,
  PressMediaPage,
  SubServicePage,
  ContactUsPage,
  OfficePage,
  NewsletterSubscriptionPage,
  PeoplePage,
  ThankYouPage,
  SearchResultsPage,
  EventsListingPage,
  WebinarListingPage,
  ContactUsSectorServicePage,
} from '.'

export interface PageBase {
  id: string
  url_en: string
  url_fr: string
  url_es?: string
  url_pt?: string
  url_de?: string
  contentTypeAlias: string
  pageTitle?: string
  videoSrc?: string
  introductionText: string
  image?: Media
  relatedContentTag: string[]
  relatedContentTypes: string[]
  navigation: Navigation[]
  footer: Footer
}

export enum PageTypeNames {
  ABOUT_PAGE = 'AboutPage',
  ALL_NEWS_PAGE = 'AllNews',
  CAREERS_PAGE = 'CareersPage',
  CONTINENT_PAGE = 'ContinentPage',
  COUNTRY_PAGE = 'CountryPage',
  CSR_PAGE = 'CSRPage',
  EVENT_DETAIL_PAGE = 'EventDetailPage',
  WEBINAR_PAGE = 'WebinarPage',
  HOME_PAGE = 'HomePage',
  INSIGHTS_LISTING_PAGE = 'InsightsListing',
  INSIGHT_PAGE = 'InsightPage',
  NEWS_DETAIL_PAGE = 'NewsDetailPage',
  PDF_DOWNLOAD_PAGE = 'PDFDownloadPage',
  PROJECT_LISTING_PAGE = 'ProjectListingPage',
  PROJECT_PAGE = 'ProjectPage',
  SECTOR_PAGE = 'SectorPage',
  SERVICES_PAGE = 'ServicesPage',
  SUB_SECTOR_PAGE = 'SubSectorPage',
  SUB_SERVICE_PAGE = 'SubServicePage',
  STANDARD_PAGE = 'StandardContentPage',
  PRESSMEDIA_PAGE = 'PressMediaPage',
  CONTACT_US_PAGE = 'ContactUs',
  OFFICE_PAGE = 'OfficePage',
  NEWSLETTER_SUBSCRIPTION_PAGE = 'NewsletterPage',
  PEOPLE_PAGE = 'PeoplePage',
  THANK_YOU_PAGE = 'ThankYouPage',
  SEARCH_RESULTS_PAGE = 'SearchResults',
  EVENT_LISTING_PAGE = 'EventsListingPage',
  WEBINAR_LISTING_PAGE = 'WebinarListingPage',
  CONTACT_US_SECTOR_SERVICE_PAGE = 'ContactUsSectorService',
}

export interface CmsRouteItem {
  __typename: PageTypeNames
  url: string
  url_fr: string
  url_es?: string
  url_pt?: string
  id: string
  updateDate: string
  hideFromXMLSitemap: boolean
  searchEngineChangeFrequency: string
  searchEngineRelativePriority: string
}

export interface CmsRoute {
  [key: string]: {
    items: CmsRouteItem[]
  }
}

export interface NavigationItem {
  __typename?: PageTypeNames | string
  navItemName: { __typename: string; name: string; target?: any; url: string }
  text?: string
  title?: string
  groups: Groups[]
}

interface Groups {
  __typename: string
  extraSpacing: boolean
  pages: Page[]
  title: string
}

interface Page {
  __typename: string
  name: string
  url: string
}

export interface Navigation {
  primaryLinks: NavigationItem[]
  secondaryLinks: NavigationItem[]
}

export interface Footer {
  subscribeLink: SubscribeLink
  primaryNavigationLinks: SubscribeLink[]
  subscribeText: string
  secondaryFooterNavigation: SubscribeLink[]
  copyrightText: string
  socialFollow: { socialPlatform: string; uRL: string }[]
  socialFollowTitle: string
}

interface SubscribeLink {
  name: string
  url: string
}

export type CmsPage =
  | AboutPage
  | AllNewsPage
  | CareerPage
  | ContinentPage
  | CountryPage
  | CSRPage
  | EventDetailPage
  | WebinarPage
  | HomePage
  | InsightsListingPage
  | InsightPage
  | NewsDetailPage
  | PDFDownloadPage
  | ProjectListingPage
  | ProjectPage
  | SectorPage
  | ServicesPage
  | SubSectorPage
  | SubServicePage
  | StandardPage
  | PressMediaPage
  | ContactUsPage
  | OfficePage
  | NewsletterSubscriptionPage
  | PeoplePage
  | ThankYouPage
  | SearchResultsPage
  | EventsListingPage
  | WebinarListingPage
  | ContactUsSectorServicePage

export interface CmsContentPage<T = CmsPage> {
  content: T
}

export type CorePageProperties = PageBase & SiteSettings & PageMetadata
