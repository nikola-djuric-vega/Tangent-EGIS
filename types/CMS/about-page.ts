import { CorePageProperties, Media, PageTypeNames } from '.'

export interface AboutPage extends CorePageProperties {
  __typename: PageTypeNames
  name: string
  url: string
  image?: Media
  introText: any
  components?: any
  contactLink?: { name: string; url: string; target: string | null }
}
