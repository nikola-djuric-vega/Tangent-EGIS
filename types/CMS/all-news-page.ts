import { CorePageProperties, Media, NewsDetailPage, PageTypeNames } from '.'

export interface AllNewsPage extends CorePageProperties {
  __typename: PageTypeNames
  name: string
  url: string
  image?: Media
  introText: any
  cTAComponent: any
  locationsLabel?: string
  newestTitle?: string
  oldestTitle?: string
  resetLabel?: string
}
