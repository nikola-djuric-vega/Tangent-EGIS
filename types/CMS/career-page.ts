import { CorePageProperties, Media, PageTypeNames } from '.'

export interface CareerPage extends CorePageProperties {
  __typename: PageTypeNames
  name: string
  url: string
  image?: Media
  date: string
  startTime: string
  introText: any
  endTime: string
  eventLabel: string
  registerLink?: { url: string; name: string }
  richText: string
  socialNetworks: []
  components: any
  parent?: { name: string; url: string }
  link?: { name: string; url: string }
}
