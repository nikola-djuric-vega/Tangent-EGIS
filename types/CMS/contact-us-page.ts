import { CorePageProperties, Media, OfficeAccordion, PageTypeNames } from '.'

export interface ContactUsPage extends CorePageProperties {
  __typename: PageTypeNames
  name: string
  url: string
  image?: Media
  introText: any
  office: OfficeAccordion[]
  headline: string
  address: string[]
  getInTouchLink: {
    __typename: string
    name: string
    url: string
  }
  contactNumber: string
}
