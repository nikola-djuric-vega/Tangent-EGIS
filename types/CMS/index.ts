export * from './about-page'
export * from './all-news-page'
export * from './base'
export * from './career-page'
export * from './csr-page'
export * from './continent-page'
export * from './country-page'
export * from './event-detail-page'
export * from './webinar-page'
export * from './home-page'
export * from './insights-listing-page'
export * from './insight-page'
export * from './link'
export * from './media'
export * from './meta-data'
export * from './news-detail-page'
export * from './office-accordion'
export * from './pdf-download-page'
export * from './project-listing-page'
export * from './project-page'
export * from './sector-page'
export * from './services-page'
export * from './sub-sector-page'
export * from './sub-service-page'
export * from './site-settings'
export * from './social-network'
export * from './standard-page'
export * from './office-accordion'
export * from './pdf-download-page'
export * from './career-page'
export * from './csr-page'
export * from './press-media-page'
export * from './contact-us-page'
export * from './form'
export * from './office-page'
export * from './newsletter-subscription-page'
export * from './error-page'
export * from './people-page'
export * from './thank-you-page'
export * from './search-results-page'
export * from './event-listing-page'
export * from './webinar-listing-page'
export * from './contact-us-sector-service-page'
