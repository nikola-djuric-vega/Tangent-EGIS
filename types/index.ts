export interface SvgProps {
  width?: number
  height?: number
  className?: string
  colour?: string
}
