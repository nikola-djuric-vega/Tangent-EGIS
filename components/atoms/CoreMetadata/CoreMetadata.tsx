import Head from 'next/head'
import { CmsPage } from '../../../types/CMS'
import { transformUrl } from '../../../utils/helpers'

export interface CoreMetadataProps {
  data: CmsPage
}

const CoreMetadata = ({ data }: CoreMetadataProps) => {
  return (
    <Head>
      {data.seoTitle && <title>{data.seoTitle}</title>}
      {data.seoDescription && (
        <meta name="description" content={data.seoDescription} />
      )}
      <link rel="icon" href="/favicon.ico" />
      {!!data.hideFromSearchEngines && <meta name="robots" content="noindex" />}
      {!!data.oGTitle && (
        <>
          <meta property="og:title" content={data.oGTitle} />
          <meta name="twitter:title" content={data.oGTitle} />
        </>
      )}
      {!!data.oGDescription && (
        <>
          <meta property="og:description" content={data.oGDescription} />
          <meta name="twitter:description" content={data.oGDescription} />
        </>
      )}
      {!!data.oGImage ? (
        <meta property="og:image" content={data.oGImage.url} />
      ) : (
        <meta
          property="og:image"
          content={require('../../../public/images/egis-logo.png').default.src}
        />
      )}

      {!!data.oGType && <meta property="og:type" content={data.oGType} />}
      {!!data.oGUrl && (
        <meta property="og:url" content={transformUrl(data.oGUrl.url)} />
      )}
      {!!data.oGLocale && <meta property="og:locale" content={data.oGLocale} />}
      {!!data.oGLocaleAlternate?.length && (
        <meta
          property="og:locale:alternate"
          content={data.oGLocaleAlternate?.join?.(',')}
        />
      )}

      <meta name="twitter:card" content="summary" />
    </Head>
  )
}

export default CoreMetadata
