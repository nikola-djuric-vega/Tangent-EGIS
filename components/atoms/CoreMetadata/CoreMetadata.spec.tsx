import React from 'react'
import { render, screen, waitFor } from '@testing-library/react'

import CoreMetadata from './CoreMetadata'
import { CmsPage, LinkType, PageTypeNames } from '../../../types/CMS'

jest.mock('next/head', () => {
  return {
    __esModule: true,
    default: function Head({
      children,
    }: {
      children: Array<React.ReactElement>
    }) {
      return <>{children}</>
    },
  }
})

const mockData: CmsPage = {
  __typename: PageTypeNames.HOME_PAGE,
  name: 'Home Page',
  url: '/home-page/',
  seoTitle: 'Egis Homepage',
  seoDescription: 'Egis Meta Description',
  hideFromSearchEngines: false,
  oGTitle: 'Egis',
  oGType: 'homepage',
  oGImage: {
    name: 'image',
    createDate: '',
    id: '',
    level: 1,
    mediaTypeAlias: 'image',
    sortOrder: 1,
    url: 'https://image.img',
  },
  oGDescription: 'Egis OG description',
  oGUrl: {
    url: '/home-page/',
    name: 'Home page',
    linkType: LinkType.CONTENT,
  },
  oGLocale: 'en_US',
  oGLocaleAlternate: ['fr'],
  pageTitle: '',
  introductionText: '',
  relatedContentTag: [],
  relatedContentTypes: [],
  id: '43fb8d0e-ddf0-48ef-8296-07059f9293ba',
  contentTypeAlias: 'homePage',
}

describe('CoreMetadata', () => {
  it('should render metadata title', async () => {
    const { baseElement } = render(<CoreMetadata data={mockData} />)

    await waitFor(() => {
      expect(baseElement).toMatchSnapshot()
    })
  })
})
