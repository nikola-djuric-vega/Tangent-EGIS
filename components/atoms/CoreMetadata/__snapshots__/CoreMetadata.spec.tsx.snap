// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreMetadata should render metadata title 1`] = `
<body>
  <div>
    <title>
      Egis Homepage
    </title>
    <meta
      content="Egis Meta Description"
      name="description"
    />
    <link
      href="/favicon.ico"
      rel="icon"
    />
    <meta
      content="Egis"
      property="og:title"
    />
    <meta
      content="Egis"
      name="twitter:title"
    />
    <meta
      content="Egis OG description"
      property="og:description"
    />
    <meta
      content="Egis OG description"
      name="twitter:description"
    />
    <meta
      content="https://image.img"
      property="og:image"
    />
    <meta
      content="homepage"
      property="og:type"
    />
    <meta
      content="/"
      property="og:url"
    />
    <meta
      content="en_US"
      property="og:locale"
    />
    <meta
      content="fr"
      property="og:locale:alternate"
    />
    <meta
      content="summary"
      name="twitter:card"
    />
  </div>
</body>
`;
